NIPTool/server/templates/statistics.html

```diff
@@ -0,0 +1,133 @@
+        {% for plot in scatter_plots %}
+        <div class="tab-pane fade in active" id='{{plot}}'>
+            <div id="{{plot}}" style="min-width:1000px; height:600px; "></div>
+        </div>
+        {% endfor %}
+    </div>
+</div>
+{% endblock %}
+
+{% block scripts %}
+
+<script>
+
+    ////////////////// Box Plots //////////////////
+    {% for plot_name in box_plots %}
+    var XY_plot = document.getElementById('')
+    var data = []
+    var layout = {
+        title: '{{plot_name}} - ({{nr_batches}} most recent batches)',
+        hovermode: 'closest',
+        margin: { b: 100 },
+        xaxis: {
+            showline: true,
+            tickvals: {{ ticks }},
+            ticktext: {{ batch_ids | tojson }},
+            tickangle: 40,
+            zeroline: false,
+            linecolor: '#636363',
+            linewidth: 5,
+            gridcolor: '#bdbdbd'},
+        yaxis: {
+            zeroline: false,
+            showline: true,
+            showgrid: false,
+            linecolor: '#636363',
+            linewidth: 5,
+            title : "{{plot_name}}"
+            }};
+
+    {% for batch in box_stat %}
+    var box = {
+        y: {{ batch[plot_name] }},
+        x: Array({{ nr_batches }}).fill({{ loop.index }}),
+        type: 'box',
+        showlegend: false,
+        boxpoints: 'suspectedoutliers',
+        };
+    data.push(box);
+    {% endfor %}
+
+    Plotly.newPlot("{{plot_name}}", data, layout);
+    {% endfor %}
+</script>
+
+<script>
+    jQuery(function () {
+        jQuery('#myTab a:first').tab('show')
+    })
+</script>
+
+<script>
+    ////////////////// Scatter plots //////////////////
+    {% for plot_name in scatter_plots %}
+    var XY_plot = document.getElementById('')
+    var data = []
+
+    var layout = {
+            title: '{{plot_name}} - ({{nr_batches}} most recent batches)',
+            hovermode: 'closest',
+            margin: { b: 100 },
+            xaxis: {
+                showline: true,
+                tickvals: {{ ticks }},
+                ticktext: {{ batch_ids| tojson }},
+                tickangle: 40,
+                zeroline: false,
+                linecolor: '#636363',
+                linewidth: 5,
+                gridcolor: '#bdbdbd',},
+            yaxis: {
+                zeroline: false,
+                showline: true,
+                showgrid: false,
+                linecolor: '#636363',
+                linewidth: 5,
+                title : "{{plot_name}}"}
+                }
+
+
+    {% for batch_id in scatter_stat %}
+        var batch = {
+            y: [{{ scatter_stat[batch_id][plot_name] }}],
+            x: [{{ loop.index  }}],
+            type: 'scatter',
+            mode: 'markers',
+            showlegend: false,
+            name: "{{batch_id}}",
+            text: "Date Run: {{scatter_stat[batch_id]['date']}}"};                          
+        data.push(batch);
+    {% endfor %}
+
+    Plotly.newPlot("{{plot_name}}", data, layout);
```

Comment:
interesting! I guess this is a way of passing variables with Jinja, is that why you don't use directly plot_name instead of "{{plot_name}}"?

Reply:
Yes exactly!